java/arcs/core/storage/CapabilitiesResolver.kt

```diff
@@ -13,25 +13,64 @@ package arcs.core.storage
 
 import arcs.core.common.ArcId
 import arcs.core.data.Capabilities
+import arcs.core.data.Schema
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
 import arcs.core.util.TaggedLog
 
 /**
  * [CapabilitiesResolver] is a factory class that creates [StorageKey]s based on the registered
  * providers and given [Capabilities].
  */
 class CapabilitiesResolver(
-    val options: StorageKeyOptions,
+    val options: CapabilitiesResolverOptions,
     private val creators: StorageKeyCreatorsMap = getAllCreators()
 ) {
     private val log = TaggedLog { "CapabilitiesResolver" }
 
     /* Options used to construct [CapabilitiesResolver] */
-    data class StorageKeyOptions(val arcId: ArcId)
+    data class CapabilitiesResolverOptions(val arcId: ArcId)
+
+    /**
+     * Options passed to registered [StorageKey] constructors.
+     * @property arcId An identifier of an Arc requesting the [StorageKey]
+     * @property entitySchema A schema of an entities that will be stored
+     * @property unique A unique component of the [StorageKey]
+     * @property location A memory location of the [StorageKey]
+     */
+    interface StorageKeyOptions {
+        val arcId: ArcId
+        val entitySchema: Schema
+        val unique: String
+            get() = TODO("not implemented")
+        val location: String
+            get() = TODO("not implemented")
+    }
+
+    data class ContainerStorageKeyOptions(
+        override val arcId: ArcId,
+        override val entitySchema: Schema
+    ) : StorageKeyOptions {
+        override val unique: String
+            get() = ""
+        override val location: String
+            get() = arcId.toString()
```

Comment:
Is it okay to use an empty value for `unique`? 

Also, as a nit: since arcId is final - you can just do 
```kotlin
override val location: String = arcId.toString()
```

Reply:
volatile and ramdisk are constructed with an empty unique - this is like a key to a base memory space, and the handle child key (with a handleId) key is truly unique.